local_cluster/src/local_cluster.rs

```diff
@@ -117,8 +116,6 @@ impl LocalCluster {
     }
 
     pub fn new(config: &ClusterConfig) -> Self {
-        set_thread_count(1);
```

Comment:
You just undid the rayon workaround. It'll start flaking again

Reply:
It seemed to work fine with release + without this workaround, I can run it a few more times to be sure.

Reply:
if you think we dont need the work around (suspicious) then rip it out fully?